### PR TITLE
[harness-automation] add another PDU Controller for EATON by SNMP

### PR DESCRIPTION
**Pull request by doublemis1**
Another PDU Controller used to power cycle between tests. Based by SNMP protocol.
To use EATON_PDU_CONTROLLER is needed PySNMP module (added to `requirements.txt`)
Also extend `settings_sample.py` by comments with needed parameters to create EATON_PDU_CONTROLLER.

### CODE REVIEW

**Comment by jwhui** [comment]
@doublemis1, will you be able to address the CLA requirement noted in https://github.com/openthread/openthread/pull/4688#issuecomment-599588229?

**Comment by doublemis1** [comment]
@googlebot I signed it.

**Comment by doublemis1** [comment]
@googlebot I fixed it.

File: tools/harness-automation/autothreadharness/pdu_controller.py [hunk 1]
```diff
@@ -189,6 +196,218 @@ def close(self):
...
+    PDU_COMMAND_TIMEOUT = 5
+
+    def open(self, **params):
+        missing_fields = ['ip', 'port']
+        missing_fields = [
+            field for field in missing_fields if field not in params.keys()
+        ]
+        if missing_fields:
+            raise KeyError(
+                "Missing keys in PDU params: {}".format(missing_fields))
```
Comment (FiveDimensions) [inline tools/harness-automation/autothreadharness/pdu_controller.py]: Suggest to use single quotes to quote strings in this and followings for being consistent with the previous program format
```suggestion
                'Missing keys in PDU params: {}'.format(missing_fields))
```

Comment (doublemis1) [reply tools/harness-automation/autothreadharness/pdu_controller.py]: Done

File: tools/harness-automation/autothreadharness/settings_sample.py [hunk 2]
```diff
@@ -122,6 +122,7 @@
     - 'APC_PDU_CONTROLLER' - when APC PDU controller connected
     - 'NORDIC_BOARD_PDU_CONTOLLER' - when Nordic boards PDU controller connected
     - 'IP_POWER_SOCKET_PDU_CONTROLLER' - when IP Power Socket 5G10A connected
+    - 'EATON_PDU_CONTROLLER' - when EATON ODU controller connected
```
Comment (FiveDimensions) [inline tools/harness-automation/autothreadharness/settings_sample.py]: typo ODU?
```suggestion
    - 'EATON_PDU_CONTROLLER' - when EATON PDU controller connected
```

Comment (jwhui) [reply tools/harness-automation/autothreadharness/settings_sample.py:125]: @doublemis1, can you confirm whether this is a typo?

**Comment by jwhui** [comment]
@doublemis1, thanks for updating. Can you also resolve the LGTM.com alerts flagged in https://github.com/openthread/openthread/pull/4688#issuecomment-607080870?

**Comment by jwhui** [comment]
> @doublemis1, thanks for updating. Can you also resolve the LGTM.com alerts flagged in [#4688 (comment)](https://github.com/openthread/openthread/pull/4688#issuecomment-607080870)?

@doublemis1, are you able to help resolve the issues flagged by LGTM.com?

**Comment by jwhui** [comment]
@doublemis1 , thanks for this contribution!

**Pull request merged**